client/client.go

```diff
@@ -759,7 +759,17 @@ func (c *Client) clearPendingOp(op *spb.AFTResult) (*OpResult, error) {
 	if v == nil {
 		return nil, fmt.Errorf("could not dequeue operation %d, unknown operation", op.Id)
 	}
-	delete(c.qs.pendq.Ops, op.Id)
+	
+	switch o := op.GetStatus() {
+	  case spb.AFTResult_FIB_PROGRAMMED, spb.AFT_FIB_FAILED:
+	     delete(q, op.id)
+	  case spb.AFTResult_RIB_PROGRAMMED:
+	     if c.state.SessParams.GetAckType() != spb.SessionParameters_RIB_AND_FIB_ACK {
+		delete(q, op.id)
+	     }
+	  case spb.AFTResult_FAILED:
```

Comment:
I think this case can be merged with `case spb.AFTResult_FIB_PROGRAMMED, spb.AFT_FIB_FAILED:`?

Reply:
for session params, there has ate_failed, fib_programmed, fib_failed, rib_programmed, and rib_failed, so i think it will be good to separate all those situations